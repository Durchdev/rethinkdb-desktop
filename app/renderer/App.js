import React, { Component } from 'react'
import { hot } from 'react-hot-loader'
import { Router } from 'react-router'
import createHashHistory from 'history/createHashHistory'
import Routes from './routes'
import StatsProvider from './contexts/StatsContext'
import './components/Icon/icons'
import './style/app.js'

const history = createHashHistory()

class App extends Component {
  render () {
    return (
      <StatsProvider>
        <Router history={history}>
          <Routes />
        </Router>
      </StatsProvider>
    )
  }
}

export default hot(module)(App)
