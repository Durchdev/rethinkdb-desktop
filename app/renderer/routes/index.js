import React from 'react'
import ErrorBoundary from 'react-error-boundary'
import Error from '../components/error'
import { Switch, Route } from 'react-router'
import Home from '../views/Home'
import Index from '../views/Connection/NewConnection'

export default () => (
  <ErrorBoundary FallbackComponent={Error}>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/newConnection" component={Index} />
    </Switch>
  </ErrorBoundary>
)
