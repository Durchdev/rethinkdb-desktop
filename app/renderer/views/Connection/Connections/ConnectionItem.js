import React from 'react'
import randomColor from 'randomcolor'
import { StyledConnectionItem } from './styles'

const ConnectionItem = props => {
  const { id, name } = props.item
  const color = randomColor({ seed: id, luminosity: 'bright' })
  return <StyledConnectionItem color={color}>{name}</StyledConnectionItem>
}

export default ConnectionItem
