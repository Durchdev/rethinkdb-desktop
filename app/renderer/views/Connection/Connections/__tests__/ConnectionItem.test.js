import React from 'react'
import ConnectionItem from '../ConnectionItem'
import renderer from 'react-test-renderer'

jest.mock('../styles', () => {
  return {
    StyledConnectionItem: ''
  }
})

test('ConnectionList render items', () => {
  const connection = { id: 1, name: 'test' }
  const component = renderer.create(<ConnectionItem item={connection} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
