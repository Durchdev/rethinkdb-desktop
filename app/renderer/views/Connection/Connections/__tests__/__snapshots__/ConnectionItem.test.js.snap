// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionList render items 1`] = `
<
  color="#fff"
>
  test
</>
`;
